code/game/machinery/vending.dm

```diff
@@ -3369,6 +3372,15 @@ var/global/num_vending_terminals = 1
 	to_chat(user, "<span class='warning'>The machine requires an ID to unlock it.</span>")
 	return 0
 
+/obj/machinery/vending/sale/update_vicon()
+	if(stat & (BROKEN))
+		src.icon_state = "sale-broken"
+	else if (stat & (NOPOWER) || custom_stock.len == 0)
```

Comment:
I think this will not be 0 if sold out, the item will still be listed, simply not in stock so it's part of the custom_stock list

Also, since you're overriding this, make sure this doesn't cause the maintenance hatch open sprite to not appear anymore

Reply:
I tested it, when all items are sold out the machine turns 'off'. The maintenance hatch sprite is handled by update_icon which I didn't touch

Reply:
Okay, great then